src/components/address-output/_macro.spec.js

```diff
@@ -5,94 +5,121 @@ import * as cheerio from 'cheerio';
-
-        const results = await axe($.html());
-        expect(results).toHaveNoViolations();
+import { EXAMPLE_ADDRESS_OUTPUT_FULL, EXAMPLE_ADDRESS_OUTPUT_NONE } from './_test_examples';
+
+describe('FOR: address-output', () => {
+    describe('GIVEN: Params: none', () => {
+        describe('WHEN: All params are at default state', () => {
+            const $ = cheerio.load(renderComponent('address-output', EXAMPLE_ADDRESS_OUTPUT_FULL));
+
+            test('THEN: jest-axe tests pass', async () => {
+                const results = await axe($.html());
+                expect(results).toHaveNoViolations();
+            });
+        });
     });
 
-    it('has additionally provided container style classes', () => {
-        const $ = cheerio.load(
-            renderComponent('address-output', {
-                ...EXAMPLE_ADDRESS_OUTPUT_FULL,
-                classes: 'extra-class another-extra-class',
-            }),
-        );
-
-        expect($('.ons-address-output').hasClass('extra-class')).toBe(true);
-        expect($('.ons-address-output').hasClass('another-extra-class')).toBe(true);
+    describe('GIVEN: Params: classes', () => {
+        describe('WHEN: classes are provided', () => {
+            const $ = cheerio.load(
+                renderComponent('address-output', {
+                    ...EXAMPLE_ADDRESS_OUTPUT_FULL,
+                    classes: 'extra-class another-extra-class',
+                }),
+            );
+
+            test('THEN: renders additionally provided classes', async () => {
+                expect($('.ons-address-output').hasClass('extra-class')).toBe(true);
+                expect($('.ons-address-output').hasClass('another-extra-class')).toBe(true);
+            });
+        });
     });
 
-    it('renders no lines when no parameters are provided', () => {
-        const $ = cheerio.load(renderComponent('address-output', EXAMPLE_ADDRESS_OUTPUT_NONE));
+    describe('GIVEN: Params: none', () => {
+        describe('WHEN: no address lines are provided in the parameters', () => {
+            const $ = cheerio.load(renderComponent('address-output', EXAMPLE_ADDRESS_OUTPUT_NONE));
 
-        expect($('.ons-address-output__lines *').length).toBe(0);
+            test('THEN: renders no lines', () => {
+                expect($('.ons-address-output__lines *').length).toBe(0);
+            });
+        });
     });
 
-    it.each([
+    describe.each([
         ['all address lines', EXAMPLE_ADDRESS_OUTPUT_FULL],
         ['single line', { unit: 'Unit 5' }],
-    ])('renders `unit` with %s', (_, params) => {
-        const $ = cheerio.load(renderComponent('address-output', params));
-
-        expect($('.ons-address-output__unit').text().trim()).toBe('Unit 5');
+    ])('GIVEN: %s', (scenario, params) => {
+        describe(`WHEN: the address-output with ${scenario}`, () => {
+            const $ = cheerio.load(renderComponent('address-output', params));
```

Comment:
The text that this produces is confusing about what it is testing

<img width="359" alt="Screenshot 2024-08-22 at 13 30 55" src="https://github.com/user-attachments/assets/9b1b7917-bf56-4c8d-829c-0041b5027ec9">

Reply:
Makes sense. Now tried to expand it better. Can you tell me if this is okay?
![Screenshot 2024-08-22 at 16 06 20](https://github.com/user-attachments/assets/6e598670-17ab-4ca3-906f-d477b196f0b1)

Reply:
Its a bit repetitive 
Maybe something like
GIVEN Params: All address line params are provided
WHEN the address is output
THEN ....